src/v2.ts

```diff
@@ -0,0 +1,153 @@
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+import {
+  CloudFunction,
+  CloudEvent,
+} from 'firebase-functions/v2';
+
+/** A function that can be called with test data and optional override values for {@link CloudEvent}
+ * It will subsequently invoke the cloud function it wraps with the provided {@link CloudEvent}
+ */
+export type WrappedV2Function = (
+  cloudEvent?: CloudEvent
+) => any | Promise<any>;
+
+/**
+ * Takes a v2 cloud function to be tested, and returns a {@link WrappedV2Function}
+ * which can be called in test code.
+ */
+export function wrapV2<T>(
+  cloudFunction: CloudFunction<T>
+): WrappedV2Function {
+
+  if (
+    // @ts-ignore
+    !!(cloudFunction?.__trigger?.httpsTrigger) &&
+    // @ts-ignore
+    cloudFunction?.__trigger?.labels?.['deployment-callable'] !== 'true'
+  ) {
+    throw new Error(
+      'Wrap function is only available for `onCall` HTTP functions, not `onRequest`.'
+    );
+  }
+
+  if (!cloudFunction.run) {
+    throw new Error(
+      'This library can only be used with functions written with firebase-functions v3.20.0 and above'
+    );
+  }
+
+  if (cloudFunction?.__endpoint?.platform !== 'gcfv2') {
+    throw new Error('This function can only wrap V2 CloudFunctions.');
+  }
+
+  return (cloudEvent: CloudEvent) => cloudFunction.run(cloudEvent);
+}
+
+/**
+ * @param cloudFunction Populates default values of the CloudEvent
+ * @param {Partial<CloudEvent>} cloudEventOverride Used to override CloudEvent params.
+ * @return {CloudEvent} Generated Mock CloudEvent
+ */
+export const createMockCloudEvent = <FunctionType, EventType>(
+  cloudFunction: CloudFunction<FunctionType>,
+  cloudEventOverride?: Partial<CloudEvent>): CloudEvent<EventType> => {
+  return {
+    ...createCloudEventWithDefaultValues<FunctionType, EventType>(cloudFunction),
+    ...cloudEventOverride,
+  } as CloudEvent<EventType>;
+};
+
+/** @internal */
+
+const getCloudEventSource =
+  <T>(cloudFunction: CloudFunction<T>): string => {
+  const projectId = '__PROJECT_ID__';
+  const type = getCloudEventType<T>(cloudFunction);
+  switch (type) {
+      case 'google.cloud.storage.object.v1.archived': // Fall-through intended
+      case 'google.cloud.storage.object.v1.deleted': // Fall-through intended
+      case 'google.cloud.storage.object.v1.finalized': // Fall-through intended
+      case 'google.cloud.storage.object.v1.metadataUpdated':
+        const bucketId = cloudFunction?.__endpoint?.eventTrigger?.eventFilters?.bucket || '';
```

Comment:
`|| '__BUCKET_ID__'`

We're also going to have a fun time once this starts being a param 😈